### ISSUE TITLE
Blocking should not break a thread

### ISSUE THREAD

**Issue by bobowen** [bluesky-social/social-app#6114]
### Describe the Feature

Currently if there is a discussion on BlueSky where people are engaging civilly and one person is responding with abuse.
The person responding with abuse can block the first person they responded to and effectively kill the whole thread.
It is somewhat reasonable that their posts are then hidden for everyone else (even people who don't block anyone in the discussion), but they should not be able to break the whole thread as I think this can lead to toxic behaviour being rewarded.

For example I have seen people finishing their discussion with a final volley of abuse and then blocking, which then breaks a whole thread of conversation when they were the only person responding with abuse.

**Comment by nielsle** [comment]
I would like to keep he present behavior.

If I start a conversation with a user that is likely to be blocked by many users, then I know that my post will be hidden for many users. That is a feature, not a bug.

**Comment by bobowen** [comment]
> I would like to keep he present behavior.
> 
> If I start a conversation with a user that is likely to be blocked by many users, then I know that my post will be hidden for many users. That is a feature, not a bug.

The problem is it can also be used for bad purposes, as I described.
Person A replies to a post ... there is a long discussion, possibly involving several people below that reply, where person A becomes frustrated and abusive ... person A then blocks the original person they replied to and possibly others, effectively destroying the whole thread and hiding their abusive behaviour.

**Comment by nielsle** [comment]
If person A blocks he original poster, then the thread is hidden for person A, but the thread is still visible for everybody else.

**Comment by bobowen** [comment]
No, if a discussion happens in replies to person A's reply then when they block it effectively kills that part of the thread.
The thread could remain visible around the blocked posts, but it doesn't.
It could also remain visible to people who follow neither person, but it doesn't.
